src/libcollections/str.rs

```diff
@@ -532,48 +593,129 @@ impl str {
     }
 
     /// Divide one mutable string slice into two at an index.
+    ///
+    /// The argument, `mid`, should be a byte offset from the start of the
+    /// string. It must also be on the boundary of a UTF-8 code point.
+    ///
+    /// The two slices returned go from the start of the string slice to `mid`,
+    /// and from `mid` to the end of the string slice.
+    ///
+    /// To get immutable string slices instead, see the [`split_at()`] method.
+    ///
+    /// [`split_at()`]: #method.split_at
+    ///
+    /// # Panics
+    ///
+    /// Panics if `mid` is not on a UTF-8 code point boundary, or if it is
+    /// beyond the last code point of the string slice.
+    ///
+    /// # Examples
+    ///
+    /// Basic usage:
+    ///
+    /// ```
+    /// let s = "Per Martin-Löf";
+    ///
+    /// let (first, last) = s.split_at(3);
+    ///
+    /// assert_eq!("Per", first);
+    /// assert_eq!(" Martin-Löf", last);
+    /// ```
     #[inline]
     #[stable(feature = "str_split_at", since = "1.4.0")]
     pub fn split_at_mut(&mut self, mid: usize) -> (&mut str, &mut str) {
         core_str::StrExt::split_at_mut(self, mid)
     }
 
-    /// An iterator over the code points of `self`.
+    /// Returns an iterator over the `char`s of a string slice.
     ///
-    /// In Unicode relationship between code points and characters is complex.
-    /// A single character may be composed of multiple code points
-    /// (e.g. diacritical marks added to a letter), and a single code point
-    /// (e.g. Hangul syllable) may contain multiple characters.
+    /// As a string slice consists of valid UTF-8, we can iterate through a
+    /// string slice by `char`. This method returns such an iterator.
     ///
-    /// For iteration over human-readable characters a grapheme cluster iterator
-    /// may be more appropriate. See the [unicode-segmentation crate][1].
```

Comment:
I think it may be good to keep this documentation around, however. It seems important to point out unicode properties like this to ensure it's not easily confused.